### ISSUE TITLE
Find path between functions does not work

### ISSUE THREAD

**Issue by trufae** [radareorg/radare2#23697]
Seems like the xrefs rewrite broke it. we lack tests for this so it was expected i guess.

reproducer:

r2 -a arm test/bins/mach0/ls-m1

```
[0x100005240]> axt $FB
sym.func.1000047a8 0x100004814 [CALL:--x] bl sym.func.100004b34
sym.func.1000047a8 0x1000049c0 [CALL:--x] bl sym.func.100004b34
[0x100005240]> s sym.func.1000047a8
[0x1000047a8]> axt $FB
main 0x100004614 [CALL:--x] bl sym.func.1000047a8
[0x1000047a8]> s main
[0x100003a90]> axt $FB
[0x100003a90]> abp?
| abp[?] [addr]  follow basic blocks paths from $$ to `addr`
[0x100003a90]> abp 0x100005240
[0x100003a90]> abp 0x100005240 @ entry0
[0x100003a90]> abl,size/sort/inc|tail -n 5
0x1000041bc 124  1      31     0x10000424c 0x100004238  0x100003a90  0x1000078c0 0x1000078c0 0x1000078c0 0x1000078c0 0x1000078c0              0x1000078c0 0x1000078c0 0x1000078c0 0x1000078c0 0x1000078c0
0x100005b54 124  1      31     0x100005bd0              0x100005638  0x1000074a0 0x1000074c0 0x100007750 0x1000075d0                          0x1000074a0 0x1000074c0 0x100007750 0x1000075d0
0x1000042fc 132  1      33     0x1000043c4 0x100004380  0x100003a90
0x100005240 140  1      35     0x1000052d4 0x1000052cc  0x100004b34  0x1000077e0 0x1000077e0 0x1000077e0                                      0x1000077e0 0x1000077e0 0x1000077e0

[0x100003a90]> o
 3 * r-x 0x0002daa0 ls-m1
 4 - r-- 0x00002560 null://9568
 5 - rw- 0x000f0000 malloc://983040
[0x100003a90]>
```

**Comment by trufae** [comment]
i think that issue is related to the refactoring of the xrefs storage using RVec. and actually it's probably the same issue spotted in https://github.com/radareorg/radare2/issues/23096 in case you are in the mood to check another similar issue :)

**Comment by condret** [comment]
can we close this now?

**Comment by trufae** [comment]
Lets reword it to implement proper algo for abpf like it is for abp

**Comment by trufae** [comment]
a new plugin has been added, this one should work well, needs tests to confirm, but at least its modular https://github.com/radareorg/radare2/pull/24607

**Comment by trufae** [comment]
@satk0 can you give it a try?

**Comment by 0verflowme** [comment]
This was fixed https://github.com/radareorg/radare2/pull/24879
